dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_Basics/AutoScaleBasics.cs

```diff
@@ -0,0 +1,247 @@
+            Console.WriteLine($"The lifecycle state is {instance.LifecycleState}");
+        }
+    }
+}
+
+Console.WriteLine("**** List the scaling activities that have occurred for the group");
+var activities = await AutoScaleMethods.DescribeAuotoScalingActivitiesAsync(client, groupName);
+if (activities is not null)
+{
+    activities.ForEach(activity =>
+    {
+        Console.WriteLine($"The activity Id is {activity.ActivityId}");
+        Console.WriteLine($"The activity details are {activity.Details}");
+    });
+}
+
+// Display the Amazon CloudWatch metrics that have been collected.
+var metrics = await CloudWatchMethods.GetCloudWatchMetricsAsync(groupName);
+Console.WriteLine($"Metrics collected for {groupName}:");
+metrics.ForEach(metric =>
+{
+    Console.Write($"Metric name: {metric.MetricName}\t");
+    Console.WriteLine($"Namespace: {metric.Namespace}");
+});
+
+var dataPoints = await CloudWatchMethods.GetMetricStatisticsAsync(groupName);
+Console.WriteLine("Details for the metrics collected:");
+dataPoints.ForEach(detail =>
+{
+    Console.WriteLine(detail);
+});
+
+// Disable metrics collection.
+Console.WriteLine("Disabling the collection of metrics for {groupName}.");
+success = await AutoScaleMethods.DisableMetricsCollectionAsync(client, groupName);
+
+if (success)
+{
+    Console.WriteLine($"Successfully stopped metrics collection for {groupName}.");
+}
+else
+{
+    Console.WriteLine($"Could not stop metrics collection for {groupName}.");
+}
+
+// Terminate all instances in the group.
+Console.WriteLine("--- Now terminating all instances in the AWS Auto Scaling group ---");
+
+groups.ForEach(group =>
+{
+    // Only delete groups in the AWS AutoScaling group we created.
+    if (group.AutoScalingGroupName == groupName)
+    {
+        group.Instances.ForEach(async instance =>
+        {
+            await AutoScaleMethods.TerminateInstanceInAutoScalingGroupAsync(client, instance.InstanceId);
+        });
+    }
+});
+
+// Once all instances have been terminated, delete the group.
+Console.WriteLine("--- Deleting the Auto Scaling group ---");
+await AutoScaleMethods.DeleteAutoScalingGroupAsync(client, groupName);
+
+// Delete the launch template.
+var deletedLaunchTemplateName = await EC2Methods.DeleteLaunchTemplateAsync(launchTemplateId);
+
+if (deletedLaunchTemplateName == launchTemplateName)
+{
+    Console.WriteLine("Successfully deleted the launch template.");
+}
+
+Console.WriteLine("The demo is now concluded.");
+
+void DisplayDescription()
+{
+    Console.WriteLine("This code example performs the following operations:");
+    Console.WriteLine(" 1. Create an Amazon EC2 launch template.");
+    Console.WriteLine(" 2. Create an AWS Auto Scaling group.");
+    Console.WriteLine(" 3. Shows the details of the new AWS Auto Scaling group");
+    Console.WriteLine("    to show that only one instance was created.");
+    Console.WriteLine(" 4. Enables  metrics collection.");
```

Comment:
Remove extra space between "Enables" and "metrics"

Reply:
fixed